File: others/invert_binary_tree.cpp
#include <bits/stdc++.h>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if(!root)   return nullptr;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            TreeNode *cur=q.front();
            q.pop();
            TreeNode *tmp=cur->left;
            cur->left = cur->right;
            cur->right = tmp;
            if(cur->left)   q.push(cur->left);
            if(cur->right)   q.push(cur->right);
        }

        return root;
    }
};

int main(){
    TreeNode *ll = new TreeNode(1);
    TreeNode *lr = new TreeNode(3);
    TreeNode *rl = new TreeNode(6);
    TreeNode *rr = new TreeNode(9);
    TreeNode *l = new TreeNode(2,ll,lr);
    TreeNode *r = new TreeNode(7,rl,rr);
    TreeNode *root = new TreeNode(4,l,r);

    Solution *sol = new Solution;
    TreeNode *new_root = sol->invertTree(root);
    queue<TreeNode*> q;
    q.push(new_root);
    while(!q.empty()){
        TreeNode *cur = q.front();
        q.pop();
        cout << cur->val << endl;
        if(cur->left)    q.push(cur->left);
        if(cur->right)   q.push(cur->right);
    }

    return 0;
}

File: others/invert_binary_tree.py
# Definition for a binary tree node.
class TreeNode:
     def __init__(self, val=0, left=None, right=None):
         self.val = val
         self.left = left
         self.right = right


class Solution:
    def traverse(self, source, target):
        target.val = source.val
        if source.left:
            target.right = TreeNode()
            self.traverse(source.left, target.right)
        if source.right:
            target.left = TreeNode()
            self.traverse(source.right, target.left)


    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return None
        new_root = TreeNode()
        self.traverse(root, new_root)
        return new_root


if __name__ == "__main__":
    sl = TreeNode(2, TreeNode(1), TreeNode(3))
    sr = TreeNode(7, TreeNode(6), TreeNode(9))
    root = TreeNode(4, sl, sr)
    sol = Solution()
    new_root = sol.invertTree(root)
    queue = [new_root]
    while queue:
        cur = queue.pop(0)
        print(cur.val)
        if cur.left:
            queue.append(cur.left)
        if cur.right:
            queue.append(cur.right)




